openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -457,13 +473,214 @@ def set_draft_version(
+    DraftChangeLog.
+
+    This function will create DraftSideEffect entries and properly add any
+    containers that may have been affected by this draft update, UNLESS it is
+    called from within a bulk_draft_changes_for block. If it is called from
+    inside a bulk_draft_changes_for block, it will not add side-effects for
+    containers, as bulk_draft_changes_for will automatically do that when the
+    block exits.
+    """
+    if set_at is None:
+        set_at = datetime.now(tz=timezone.utc)
+
+    tx_context = atomic() if create_transaction else nullcontext()
+
+    with tx_context:
+        old_version_id = draft.version_id
+        draft.version_id = publishable_entity_version_pk
+        draft.save()
+
+        learning_package_id = draft.entity.learning_package_id
+
+        # Check to see if we're inside a context manager for an active
+        # DraftChangeLog (i.e. what happens if the caller is using the public
+        # bulk_draft_changes_for() API call), or if we have to make our own.
+        active_change_log = DraftChangeLogContext.get_active_draft_change_log(learning_package_id)
+        change_log = active_change_log or DraftChangeLog.objects.create(
+            learning_package_id=learning_package_id,
+            changed_at=set_at,
+            changed_by_id=set_by,
+        )
+
+        if active_change_log:
+            change = _add_to_existing_draft_change_log(
+                active_change_log,
+                draft.entity_id,
+                old_version_id=old_version_id,
+                new_version_id=publishable_entity_version_pk,
+            )
+            # We explicitly *don't* create container side effects here because
+            # there may be many changes in this DraftChangeLog, some of which
+            # haven't been made yet. It wouldn't make sense to create a side
+            # effect that says, "this Unit changed because this Component in it
+            # changed" if we were changing that same Unit later on in the same
+            # DraftChangeLog, because that new Unit version might not even
+            # include the child Component. So we'll let DraftChangeLogContext
+            # do that work when it exits its context.
+        else:
+            # This means there is no active DraftChangeLog, so we create our own
+            # and add our DraftChangeLogRecord to it. This has the minor
+            # optimization that we don't have to check for an existing
+            # DraftChangeLogRecord, because we know it can't exist yet.
+            change = DraftChangeLogRecord.objects.create(
+                draft_change_log=change_log,
+                entity_id=draft.entity_id,
+                old_version_id=old_version_id,
+                new_version_id=publishable_entity_version_pk,
+            )
+            _create_container_side_effects_for_draft_change(change)
+
+
+@set_draft_version.register(int)
```

Comment:
```suggestion
@set_draft_version.register
```

[Looks like you can omit `(int)` this since the first argument is type-annotated.](https://docs.python.org/3/library/functools.html#functools.singledispatch). Same for t

Reply:
I don't understand why, but it doesn't infer/dispatch properly if I don't include `(int)`. A bunch of the tests break with error messages that show that it called the base version expecting a `Draft` with an `int` argument, e.g.:

```
        tx_context = atomic() if create_transaction else nullcontext()
    
        with tx_context:
>           old_version_id = draft.version_id
E           AttributeError: 'int' object has no attribute 'version_id'
```

Reply:
Weird, gotcha. I might poke at this if I have time, but no need to block on it.